app/javascript/hyacinth_v1/components/batch_imports/BatchImports.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Route, Switch } from 'react-router-dom';
+
+import PageNotFound from '../shared/PageNotFound';
+import ProtectedRoute from '../shared/routes/ProtectedRoute';
+import BatchImportIndex from './BatchImportIndex';
+import BatchImportShow from './BatchImportShow';
+import BatchImportNew from './BatchImportNew';
+import DigitalObjectImports from './digital_object_imports/DigitalObjectImports';
+
+function BatchImports() {
+  return (
+    <Switch>
+      <Route exact path="/batch_imports" component={BatchImportIndex} />
+      <Route exact path="/batch_imports/new" component={BatchImportNew} />
+      <Route path="/batch_imports/:id/digital_object_imports" component={DigitalObjectImports} />
+
+      {/* need to check  permissions somehow */}
```

Comment:
```suggestion
      {/* TODO: need to check  permissions somehow */}
```
I just added "TODO" to make this easier to find later with other TODO notes.  Also, if this isn't ticketed yet, it would be good to create a ticket for it.

Reply:
Is the problem here that we don't have easy access to the current user's id in this context?  Maybe we want to make it more globally accessible across the client-side code?

This same type of permission check may apply to DigitalObjectImports.js routes too.

Reply:
I think that we are _not_ going to be able to successfully check permissions for BatchImports before we render the page. In order to successfully check permissions I need to be able to retrieve the actual object (BatchImport) itself in order to check what user it belongs to. In order to retrieve the object the user needs to already have permission to the object. At some point, I thought we would be able to check permissions via the user id, but then I realized that we don't have that information to do that if the user doesn't have access to the import. I think we should just remove this TODOs and depend on the graphql errors that will be raised and displayed if a user doesn't have access.

Reply:
Sounds good to me!